app/src/main/java/com/kylecorry/trail_sense/tools/metaldetector/ui/FragmentToolMetalDetector.kt

```diff
@@ -80,6 +88,29 @@ class FragmentToolMetalDetector : BoundFragment<FragmentToolMetalDetectorBinding
                 isMetalDetected.debounceTime = Duration.ofMillis(100)
             }
         }
+
+
+
+        binding.metalDetectorTitle.rightButton.setOnClickListener {
+            if (prefs.isMetalAudioEnabled){
+                prefs.isMetalAudioEnabled = false
+                whistle.off()
```

Comment:
In the onPause method, can you also turn the whistle off?

Reply:
I added logic for onResume.. to initializeWhistle() as I notice the instance is cleared on pause